### PR TITLE
Add co_refract function

### PR DESCRIPTION
**Pull request by TestSubjector**
Also includes minor function `co_refract_forward`

* TODO.md: Remove co_refract from list.
* docs/src/ref.md: Add co_refract entry to the manual.
* src/utils.jl: Add co_refract entry to "utils".
* src/co_refract.jl: Contains code of co_refract function.
* test/util-tests.jl: Include tests for co_refract.

### CODE REVIEW

File: src/co_refract.jl [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _co_refract(old_alt::T, altitude::T, pressure::T, temperature::T,
+                    epsilon::T, to_observe::Bool) where {T<:AbstractFloat}
+
+    if isnan(temperature)
+        if altitude > 11000
+            temperature = 211.5
```
Comment (giordano) [inline src/co_refract.jl]: If `T != Float64`, `temperature` will be type-unstable here.  Convert to `T` type with `T(211.5)` ;-)

File: src/co_refract.jl [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    if isnan(pressure)
+        pressure = 1010 * (1 - 6.5/288000 * altitude) ^ 5.255
+    end
+
+    if !to_observe
+        aout = old_alt - co_refract_forward(old_alt, pressure, temperature)
+    else
+        cur = old_alt + co_refract_forward(old_alt, pressure, temperature)
+        last = zero(T)
+        while true
```
Comment (giordano) [inline src/co_refract.jl]: Maybe I'm missing something, but isn't this `while` equivalent to
```julia
while abs(last-cur) * 3600 >= epsilon
    last = cur
    cur = old_alt + co_refract_forward(last, pressure, temperature)
end
```
?

Comment (TestSubjector) [inline src/co_refract.jl]: At this point, I should commit seppuku out of shame.

Comment (giordano) [inline src/co_refract.jl]: I don't think it'll be necessary ;-)  I have to admit that I didn't look to the IDL code before posting the comments, now I have to say that your solution was almost good: in IDL a [`repeat...until`](https://www.harrisgeospatial.com/docs/repeat___until.html) loop is used, which evaluates the body of the loop at least once, much like the [`do...while`](https://en.wikipedia.org/wiki/Do_while_loop) loop in C.  Unfortunately, Julia doesn't have such construct, but you can get the same effect with:
```julia
while true
    last = cur
    cur = old_alt + co_refract_forward(last, pressure, temperature)
    if abs(last-cur) * 3600 < epsilon
        break
    end
end
```
Sorry for not having noticed this earlier.

Comment (TestSubjector) [inline src/co_refract.jl]: No problem :+1:

**Pull request merged**